ResearchKit/Common/ORKLocationSelectionView.m

```diff
@@ -0,0 +1,343 @@
+        [self removeConstraint:_textFieldBottomConstraint];
+        _textFieldBottomConstraint = [NSLayoutConstraint constraintWithItem:_textField attribute:NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:_mapView attribute:NSLayoutAttributeTop multiplier:1.0 constant:0.0];
+        [self addConstraint:_textFieldBottomConstraint];
+        _mapViewHeightConstraint = [NSLayoutConstraint constraintWithItem:_mapView attribute:NSLayoutAttributeHeight relatedBy:NSLayoutRelationEqual toItem:nil attribute:NSLayoutAttributeNotAnAttribute multiplier:1.0 constant:0.0];
+        [self addConstraint:_mapViewHeightConstraint];
+        [self addConstraint:[NSLayoutConstraint constraintWithItem:_mapView attribute:NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:self attribute:NSLayoutAttributeBottom multiplier:1.0 constant:0.0]];
+        
+        [self layoutIfNeeded];
+        _mapViewHeightConstraint.constant = 228.0;
+        __weak __typeof__(self) weakSelf = self;
+        [UIView animateWithDuration:0.3 animations:^{
+            [weakSelf layoutIfNeeded];
+            
+            if ([weakSelf.delegate respondsToSelector:@selector(selectionViewNeedsResize:)]) {
+                [weakSelf.delegate selectionViewNeedsResize:self];
+            }
+        }];
+    }
+}
+
+- (void)hideMapViewAnimated:(BOOL)animated {
+    if (!_mapView) {
+        return;
+    }
+    
+    if (!animated) {
+        [_mapView removeFromSuperview];
+        _mapView = nil;
+        _mapViewHeightConstraint = nil;
+        _textFieldBottomConstraint = [NSLayoutConstraint constraintWithItem:_textField attribute:NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:self attribute:NSLayoutAttributeBottom multiplier:1.0 constant:0.0];
+        [self addConstraint:_textFieldBottomConstraint];
+        [self layoutIfNeeded];
+    } else {
+        [self layoutIfNeeded];
+        _mapViewHeightConstraint.constant = 0.0;
+        __weak __typeof__(self) weakSelf = self;
+        [UIView animateWithDuration:0.3 animations:^{
+            [weakSelf layoutIfNeeded];
+        } completion:^(BOOL finished) {
+            [weakSelf.mapView removeFromSuperview];
+            weakSelf.mapView = nil;
+            weakSelf.mapViewHeightConstraint = nil;
+            weakSelf.textFieldBottomConstraint = [NSLayoutConstraint constraintWithItem:weakSelf.textField attribute:NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:weakSelf attribute:NSLayoutAttributeBottom multiplier:1.0 constant:0.0];
+            [weakSelf addConstraint:weakSelf.textFieldBottomConstraint];
+            [weakSelf layoutIfNeeded];
+            if ([weakSelf.delegate respondsToSelector:@selector(selectionViewNeedsResize:)]) {
+                [weakSelf.delegate selectionViewNeedsResize:self];
+            }
+        }];
```

Comment:
Create a strong reference to the weakSelf inside the block because the weakSelf is at risk of being deallocated during the execution of the block.